open_spiel/fog/observation_history.cc

```diff
@@ -142,6 +142,15 @@ bool ActionObservationHistory::IsExtensionOf(Player pl,
   return CheckStateCorrespondenceInSimulation(pl, state, state.MoveNumber());
 }
 
+void ActionObservationHistory::Extend(const std::optional<Action> action, const std::string& observation_string) {
```

Comment:
Now that I think of it I think it will be better to drop these and not to track them inside of State... it's expensive and if someone wants to do it they can do it in their algorithm, as checking on what types of observations the domain provides -- i.e. that dark chess doesn't have perfect recall obs.